src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -0,0 +1,861 @@
+            result = value.toString();
+        }
+        else if (fieldClass == int[].class) {
+            result = getIntArrayPropertyValue(value);
+        }
+        else if (fieldClass == double[].class) {
+            result = removeSquareBrackets(Arrays.toString((double[]) value).replace(".0", ""));
+            if (result.isEmpty()) {
+                result = CURLY_BRACKETS;
+            }
+        }
+        else if (fieldClass == String[].class) {
+            result = getStringArrayPropertyValue(propertyName, value);
+        }
+        else if (fieldClass == URI.class || fieldClass == String.class) {
+            if (value != null) {
+                result = '"' + value.toString() + '"';
+            }
+        }
+        else if (fieldClass == Pattern.class) {
+            if (value != null) {
+                result = '"' + value.toString().replace("\n", "\\n").replace("\t", "\\t")
+                        .replace("\r", "\\r").replace("\f", "\\f") + '"';
+            }
+        }
+        else if (fieldClass == Pattern[].class) {
+            result = getPatternArrayPropertyValue(value);
+        }
+        else if (fieldClass.isEnum()) {
+            if (value != null) {
+                result = value.toString().toLowerCase(Locale.ENGLISH);
+            }
+        }
+        else if (fieldClass == AccessModifierOption[].class) {
+            result = removeSquareBrackets(Arrays.toString((Object[]) value));
+        }
+        else {
+            final String message = String.format(Locale.ROOT,
+                    "Unknown property type: %s", fieldClass.getSimpleName());
+            throw new MacroExecutionException(message);
+        }
+
+        if (result == null) {
+            result = "null";
+        }
+
+        return result;
+    }
+
+    /**
+     * Gets the name of the bean property's default value for the Pattern array class.
+     *
+     * @param fieldValue The bean property's value
+     * @return String form of property's default value
+     */
+    private static String getPatternArrayPropertyValue(Object fieldValue) {
+        Object value = fieldValue;
+        String result;
+        if (value instanceof Collection) {
+            final Collection<?> collection = (Collection<?>) value;
+            final Pattern[] newArray = new Pattern[collection.size()];
+            final Iterator<?> iterator = collection.iterator();
+            int index = 0;
+
+            while (iterator.hasNext()) {
+                final Object next = iterator.next();
+                newArray[index] = (Pattern) next;
+                index++;
+            }
+
+            value = newArray;
+        }
+
+        if (value != null && Array.getLength(value) > 0) {
+            final String[] newArray = new String[Array.getLength(value)];
+
+            for (int index = 0; index < newArray.length; index++) {
+                newArray[index] = ((Pattern) Array.get(value, index)).pattern();
+            }
```

Comment:
transformation/map of one array to another array ....
can we use some fancy streams/map ?

Reply:
Done. Simplified.